channeldb/db.go

```diff
@@ -114,11 +120,12 @@ var bufPool = &sync.Pool{
 type DB struct {
 	*bbolt.DB
 	dbPath string
+	dryRun bool
 }
 
 // Open opens an existing channeldb. Any necessary schemas migrations due to
 // updates will take place as necessary.
-func Open(dbPath string) (*DB, error) {
```

Comment:
could we rather make a new method `DryRunDB`, so we don't need to pass the `bool` all over? Internally they both can share `func open(dbPath string, dryRun bool) (*DB, error)`.

Would also make sure this commit compiles.

Reply:
now passed an optional modifier